src/uwtools/tests/drivers/test_forecast.py

```diff
@@ -34,157 +37,162 @@ def test_batch_script():
-    FV3Forecast.create_directory_structure(rundir, "rename")
-    copy_directory = next(tmp_path.glob("%s_*" % rundir.name))
-    assert (copy_directory / "RESTART").is_dir()
-
-    # Test quit behavior when run directory exists.
-    with raises(SystemExit) as pytest_wrapped_e:
-        FV3Forecast.create_directory_structure(rundir, "quit")
-    assert pytest_wrapped_e.type == SystemExit
-    assert pytest_wrapped_e.value.code == 1
-
-
-@fixture
-def create_field_table_update_obj():
-    return YAMLConfig(fixture_path("FV3_GFS_v16_update.yaml"))
-
-
-def test_create_field_table_with_base_file(create_field_table_update_obj, tmp_path):
-    """
-    Tests create_field_table method with optional base file.
-    """
-    base_file = fixture_path("FV3_GFS_v16.yaml")
-    outfldtbl_file = tmp_path / "field_table_two.FV3_GFS"
-    expected = fixture_path("field_table_from_base.FV3_GFS")
-    config_file = tmp_path / "fcst.yaml"
-    forecast_config = create_field_table_update_obj
-    forecast_config["forecast"]["field_table"]["base_file"] = base_file
-    forecast_config.dump(config_file)
-    FV3Forecast(config_file).create_field_table(outfldtbl_file)
-    assert compare_files(expected, outfldtbl_file)
-
-
-def test_create_field_table_without_base_file(tmp_path):
-    """
-    Tests create_field_table without optional base file.
-    """
-    outfldtbl_file = tmp_path / "field_table_one.FV3_GFS"
-    expected = fixture_path("field_table_from_input.FV3_GFS")
-    config_file = fixture_path("FV3_GFS_v16_update.yaml")
-    FV3Forecast(config_file).create_field_table(outfldtbl_file)
-    assert compare_files(expected, outfldtbl_file)
-
-
-def test_create_directory_structure_bad_existing_act():
-    with raises(ValueError):
-        FV3Forecast.create_directory_structure(run_directory="/some/path", exist_act="foo")
-
-
-def test_create_model_configure_call_private(tmp_path):
-    basefile = str(tmp_path / "base.yaml")
-    infile = fixture_path("forecast.yaml")
-    outfile = str(tmp_path / "out.yaml")
-    for path in infile, basefile:
-        Path(path).touch()
-    with patch.object(Driver, "_create_user_updated_config") as _create_user_updated_config:
+        config_file = fixture_path("forecast.yaml")
+        with patch.object(Driver, "_validate", return_value=True):
+            forecast = FV3Forecast(config_file=config_file)
+        assert forecast.batch_script().content() == expected
+
+
+class TestFV3Forecast:
+    def test_create_directory_structure(self, tmp_path):
+        """
+        Tests create_directory_structure method given a directory.
+        """
+
+        rundir = tmp_path / "rundir"
+
+        # Test delete behavior when run directory does not exist.
+        FV3Forecast.create_directory_structure(rundir, "delete")
+        assert (rundir / "RESTART").is_dir()
+
+        # Create a file in the run directory.
+        test_file = rundir / "test.txt"
+        test_file.touch()
+        assert test_file.is_file()
```

Comment:
Would it make sense to factor these blocks out into individual tests?  The seem kind of independent and, if they do depend on the previous blocks working, maybe they shoudln't? There could be a class-level fixture like
```
@fixture
def rundir(self, tmp_path):
    return tmp_path / "rundir"
```
I count maybe 5 potential individual test in this method. I don't feel strongly about it, just thought it was worth checking.

Reply:
Good call. I think it makes it clearer. 

I left the permissions check tied to the first creation test, but separated out the others.